pydrive2/auth.py

```diff
@@ -226,27 +190,53 @@ def LocalWebserverAuth(
         :raises: AuthenticationRejected, AuthenticationError
         """
         if port_numbers is None:
-            port_numbers = [
-                8080,
-                8090,
-            ]  # Mutable objects should not be default
-            # values, as each call's changes are global.
-        success = False
-        port_number = 0
-        for port in port_numbers:
-            port_number = port
-            try:
-                httpd = ClientRedirectServer(
-                    (host_name, port), ClientRedirectHandler
+            port_numbers = [8080, 8090]
+
+        additional_config = {}
+        # offline token request needed to obtain refresh token
+        # make sure that consent is requested
+        if self.settings.get("get_refresh_token"):
+            additional_config["access_type"] = "offline"
+            additional_config["prompt"] = "select_account"
+
+        try:
+            for port in port_numbers:
+                self._credentials = self.flow.run_local_server(
+                    host=host_name,
+                    port=port,
+                    authorization_prompt_message=_CLIENT_AUTH_PROMPT_MESSAGE,
+                    open_browser=launch_browser,
+                    **additional_config,
                 )
-            except OSError:
-                pass
-            else:
-                success = True
+                # if any port results in successful auth, we're done
                 break
-        if success:
-            oauth_callback = f"http://{host_name}:{port_number}/"
-        else:
+
+        except OSError as e:
+            # OSError: [WinError 10048] ...
+            # When WSGIServer.allow_reuse_address = False,
+            # raise OSError when binding to a used port
+
+            # If some other error besides the socket address reuse error
+            if e.errno != 10048:
```

Comment:
this is a new logic, could you clarify where it comes from please? why this error code specifically?